### DISCUSSION TITLE
Complete AST traversal instead of at a specific position

### DISCUSSION THREAD

**Discussion by kerams** [dotnet/fsharp#10721] [Q&A]
`AstTraversal.Traverse`  requires a `pos` parameter that restricts diving into nodes that don't contain the particular position. I'm looking for record usages in a document and need to look through the entire AST. Is there a way to do this right now short of reimplementing `AstTraversal`?

**Comment by cartermp** [comment]
There isn't a way to do that yet, no. You'll need to implement your own traversal or override `VisitModuleOrNamespace` and effectively write the traversal yourself anyways.

**Reply by kerams** [reply]
Not what I wanted to hear :). On a related note, is TAST exposed (or exposable) to editor tooling in a similar fashion (I realized the untyped tree does not cut it for record identification)?

**Reply by cartermp** [reply]
No, it's not exposed.
